java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java

```diff
@@ -207,6 +209,7 @@ public int getSchema(FlatBufferBuilder builder) {
     int fieldsOffset = org.apache.arrow.flatbuf.Schema.createFieldsVector(builder, fieldOffsets);
     int metadataOffset = FBSerializables.writeKeyValues(builder, metadata);
     org.apache.arrow.flatbuf.Schema.startSchema(builder);
+    org.apache.arrow.flatbuf.Schema.addEndianness(builder, (short) (LITTLE_ENDIAN ? 0 : 1));
```

Comment:
nit: we could use the generated constants for this? [org.apache.arrow.flatbuf.Endianness](https://arrow.apache.org/docs/java/reference/org/apache/arrow/flatbuf/Endianness.html)